### ISSUE TITLE
Meshes without UV2 cannot contribute to LightmapGI

### ISSUE THREAD

**Issue by sievaxx** [godotengine/godot#96538]
### Tested versions

- Tested in Godot 4.3-stable

### System information

Godot v4.3.stable - Windows 10.0.22621 - Vulkan (Forward+) - dedicated NVIDIA GeForce GTX 1080 Ti (NVIDIA; 31.0.15.3742) - Intel(R) Core(TM) i7-8700K CPU @ 3.70GHz (12 Threads)

### Issue description

Meshes that have no UV2, even when the GeometryInstance3D `gi_mode` is set to Static, do not appear to contribute any shadows to the LightmapGI. 
![image](https://github.com/user-attachments/assets/35abca56-f1c4-4b93-b4c1-3adfb0e52f13)

Unwrapping the UV2 does allow things to contribute, but to me this doesn't make any sense. They should still be capable of being rasterized by the lightmapper for direct light even if they have no UV2 to allow for writing to the lightmap, and it should be up to the `gi_mode` to dictate if it is contributing. I'd understand if they couldn't have bounce light when `use_texture_for_bounces` is true, but it should still be visible to the direct light.
![image](https://github.com/user-attachments/assets/5dce82d6-e73c-4660-a6d8-83993a4c9672)

Currently this makes setting up "light blockers", like gobos or to prevent light bleeding, annoying. You have to make sure they have UV2s, and then having a hint size small to not waste space on the atlas. There doesn't seem to be a better way to do it then that

### Steps to reproduce

Just bake a LightmapGI in a scene with meshes that have no UV2 alongside meshes that do.

### Minimal reproduction project (MRP)

N/A

**Comment by bikemurt** [comment]
The documentation mentions in several places that `UV2` is required (and hence why primitives won't work for light baking). My understanding is that the baked light textures map to UV2, and not the mesh's original UV map, which might not be using maximum surface area.

The workflow for light blockers might be the main issue here.

**Comment by sievaxx** [comment]
> The documentation mentions in several places that `UV2` is required (and hence why primitives won't work for light baking). My understanding is that the baked light textures map to UV2, and not the mesh's original UV map, which might not be using maximum surface area.

I didn't misunderstand that, meshes obviously need UV2s to be baked into the lightmap texture atlas. But meshes without UV2s should still be considered in the lightmap raycasting/rasterization for other surfaces, even if they themselves have nothing to bake onto.

**Comment by Calinou** [comment]
This was supported in Godot 3.x, but it wasn't reimplemented in Godot 4.x when the lightmapper was rewritten from scratch.

I agree this would be useful for emissive surfaces that you hide after baking (or large emissive surfaces you don't want to take space in the generated lightmap textures), but this change can change visuals in existing projects if they bake lightmaps again.

**Comment by bertodelrio256** [comment]
> This was supported in Godot 3.x, but it wasn't reimplemented in Godot 4.x when the lightmapper was rewritten from scratch. The way the GPU lightmapper works may make this not technically feasible anymore as well.
> 
> I agree this would be useful for emissive surfaces that you hide after baking (or large emissive surfaces you don't want to take space in the generated lightmap textures). However, changing this behavior may change visuals in existing projects if they bake lightmaps again.

to prevent this, why not create a new GI mode "Static  - Probe Only". that way the user would need to switch the mode to see this new behavior.

**Comment by Calinou** [comment]
> to prevent this, why not create a new GI mode "Static - Probe Only". that way the user would need to switch the mode to see this new behavior

Yes, but the Probe Only name wouldn't accurately represent what would happen (since the object would still cast shadows on lightmapped surfaces for lights with the Static bake mode).

A name like "Static (Contribute Only)" would be more suited.

**Comment by atirut-w** [comment]
> The way the GPU lightmapper works may make this not technically feasible anymore as well.

Unity can do this, so I don't see why Godot can't.

**Comment by bertodelrio256** [comment]
> > to prevent this, why not create a new GI mode "Static - Probe Only". that way the user would need to switch the mode to see this new behavior
> 
> Yes, but the Probe Only name wouldn't accurately represent what would happen (since the object would still cast shadows on lightmapped surfaces for lights with the Static bake mode).
> 
> A name like "Static (Contribute Only)" would be more suited.

agreed

**Comment by lander-vr** [comment]
> A name like "Static (Contribute Only)" would be more suited.

Though I understand the reasoning, this fix seems a little bit messy to me as the current behavior for meshes without UV2 with `gi_mode` set to `static` is the same behavior as if it'd be set to `disabled`, and the `Static (Contribute Only)` mode seems to me like the intuitive implied behavior of `static`.

If we do add a new mode for the purpose of bypassing breaking changes, I think we should definitely keep track of this somewhere so we can (and remember to) consolidate that compatibility mode into the regular static mode on the next major version bump.
